tests/withdrawals_test.go

```diff
@@ -0,0 +1,57 @@
+package tests
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithdrawalFieldsInBlocks(t *testing.T) {
+	requireBase := require.New(t)
+
+	// start network.
+	net, err := StartIntegrationTestNet(t.TempDir())
+	requireBase.NoErrorf(err, "Failed to start the fake network: ", err)
+	defer net.Stop()
+
+	// run endowment to ensure at least one block exists
+	receipt, err := net.EndowAccount(common.Address{42}, 1)
+	requireBase.NoError(err)
+	requireBase.Equal(receipt.Status, types.ReceiptStatusSuccessful, "failed to endow account")
+
+	// get client and block
+	client, err := net.GetClient()
+	requireBase.NoError(err, "Failed to get the client: ", err)
+	defer client.Close()
+	block, err := client.BlockByNumber(context.Background(), receipt.BlockNumber)
+	requireBase.NoError(err, "Failed to get the block: ", err)
+
+	t.Run("verify default values of block's Withdrawals list and hash", func(t *testing.T) {
```

Comment:
Nit: withdrawals hash is empty?

Reply:
depending on the revision, for backwards compatibility, it can be nil, but for Sonic it should never be nil, it should have a default value.